### PR TITLE
Some much needed graph api cleanup

### PR DESCRIPTION
**Pull request by fppt**
- Killing Felix Tests
- Getting rid of redundant code
- Other cleanupes

### CODE REVIEW

**Comment by fppt** [comment]
please retest this please

**Comment by fppt** [comment]
please retest this please

File: grakn-graph/src/test/java/ai/grakn/graph/internal/GraknGraphPropertyIT.java [hunk 1]
```diff
@@ -344,17 +344,18 @@ public void whenCallingSuperTypeOnMetaResourceType_Throw(@Open GraknGraph graph)
...
-            type.hasResource(resource);
-        } catch (ClassCastException e) {
-            fail();
+        exception.expect(ConceptException.class);
+        if(Schema.MetaSchema.isMetaName(type.getName())) {
+            exception.expectMessage(ErrorMessage.META_TYPE_IMMUTABLE.getMessage(type.getName()));
+        } else {
+            exception.expectMessage(ErrorMessage.META_TYPE_IMMUTABLE.getMessage(resource.getName()));
         }
+        type.hasResource(resource);
```
Comment (aelred) [inline grakn-graph/src/test/java/ai/grakn/graph/internal/GraknGraphPropertyIT.java]: Perhaps this test should be renamed?

Comment (fppt) [inline grakn-graph/src/test/java/ai/grakn/graph/internal/GraknGraphPropertyIT.java]: Done

**Pull request merged**